reanalysis/wrapper.py

```diff
@@ -0,0 +1,257 @@
+
+    :param matrix_path: path to annotated matrix table
+    :param config: can this be used as a dict if already loaded?
+    :return:
+    """
+    init_batch()
+    category_main(
+        mt_input=matrix_path,
+        mt_tmp=MT_TMP,
+        config_path=config,
+        panelapp_path=PANELAPP_JSON_OUT,
+        out_vcf=HAIL_VCF_OUT,
+    )
+
+
+def handle_reheader_job(
+    batch: hb.Batch,
+    local_vcf: str,
+    config_dict: Dict[str, Any],
+    prior_job: Optional[hb.batch.job.BashJob] = None,
+) -> hb.batch.job.BashJob:
+    """
+    runs the bcftools re-header process
+    :param batch:
+    :param local_vcf:
+    :param config_dict:
+    :param prior_job:
+    :return:
+    """
+
+    bcft_job = batch.new_job(name='bcftools_reheader_stage')
+    set_job_resources(bcft_job)
+    bcft_job.image(BCFTOOLS_IMAGE)
+
+    if prior_job is not None:
+        bcft_job.depends_on(prior_job)
+
+    bcft_job.declare_resource_group(
+        vcf={'vcf': '{root}.vcf.bgz', 'vcf.tbi': '{root}.vcf.bgz.tbi'}
+    )
+
+    # reheader the VCF using BCFtools and sed
+    # replace the empty description with the full CSQ line from config
+    desc = '##INFO=<ID=CSQ,Number=.,Type=String,Description="'
+
+    # grotty string formatting to deliver the correct syntax to bcftools
+    conf_csq = config_dict['variant_object'].get('csq_string').replace('|', r'\|')
+    new_format = rf"Format: '{conf_csq}'"
+
+    # only to reduce line length
+    b_rh = 'bcftools reheader'
+
+    bcft_job.command(
+        'set -ex; '
+        f'bcftools view -h {local_vcf} | sed \'s/'
+        f'{desc}">/{desc}{new_format}">/\' > new_header; '
+        f'{b_rh} -h new_header --threads 4 -o {bcft_job.vcf["vcf"]} {local_vcf}; '
+        f'tabix {bcft_job.vcf["vcf"]}; '
+    )
+    return bcft_job
+
+
+@click.command()
+@click.option(
+    '--matrix', 'matrix_path', help='variant matrix table to analyse', required=True
+)
+@click.option(
+    '--config_json',
+    help='dictionary of runtime settings',
+    default='gs://cpg-acute-care-test/reanalysis/reanalysis_conf.json',
+)
+@click.option(
+    '--panelapp_version',
+    help='panelapp current comparison with this earlier version',
+    required=False,
+)
+@click.option(
+    '--panel_genes',
+    help='location of a Gene list for use in analysis',
+    required=False,
+)
+def main(
+    matrix_path: str,
+    config_json: str,
+    panelapp_version: Optional[str],
+    panel_genes: Optional[str],
+):
+    """
+    main method, which runs the full reanalysis process
+
+    :param matrix_path: annotated input matrix table
+    :param config_json:
+    :param panelapp_version:
+    :param panel_genes:
+    # :param ped_file:
+    """
+
+    logging.info('Starting the reanalysis batch')
+
+    config_dict = read_json_dict_from_path(config_json)
+
+    service_backend = hb.ServiceBackend(
+        billing_project=os.getenv('HAIL_BILLING_PROJECT'),
+        bucket=os.getenv('HAIL_BUCKET'),
```

Comment:
This is deprecated. You can import `remote_tmpdir` from `cpg_utils` and instead use:
```suggestion
        remote_tmpdir=remote_tmpdir()
```